pysat/tests/classes/cls_instrument_access.py

```diff
@@ -0,0 +1,841 @@
+        return
+
+    def test_basic_instrument_load_yr_no_doy(self):
+        """Ensure day of year required if year is present."""
+
+        # Check that the correct error is raised
+        with pytest.raises(TypeError) as err:
+            self.testInst.load(self.ref_time.year)
+
+        # Check that the error message is correct
+        estr = 'Unknown or incomplete input combination.'
+        assert str(err).find(estr) >= 0
+        return
+
+    @pytest.mark.parametrize('doy', [0, 367, 1000, -1, -10000])
+    def test_basic_instrument_load_yr_bad_doy(self, doy):
+        """Ensure day of year load argument in valid range."""
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, doy)
+        estr = 'Day of year (doy) is only valid between and '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize('end_doy', [0, 367, 1000, -1, -10000])
+    def test_basic_instrument_load_yr_bad_end_doy(self, end_doy):
+        """Ensure `end_doy` keyword in valid range."""
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, 1, end_yr=self.ref_time.year,
+                               end_doy=end_doy)
+        estr = 'Day of year (end_doy) is only valid between and '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    def test_basic_instrument_load_yr_no_end_doy(self):
+        """Ensure `end_doy` required if `end_yr` present."""
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, self.ref_doy,
+                               self.ref_time.year)
+        estr = 'Both end_yr and end_doy must be set'
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize("input", [{'yr': 2009, 'doy': 1,
+                                        'date': dt.datetime(2009, 1, 1)},
+                                       {'yr': 2009, 'doy': 1,
+                                        'end_date': dt.datetime(2009, 1, 1)},
+                                       {'yr': 2009, 'doy': 1,
+                                        'fname': 'dummy_str.nofile'},
+                                       {'yr': 2009, 'doy': 1,
+                                        'stop_fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'stop_fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'fname': 'dummy_str.nofile',
+                                        'end_yr': 2009, 'end_doy': 1}])
+    def test_basic_instrument_load_mixed_inputs(self, input):
+        """Ensure mixed load inputs raise ValueError."""
```

Comment:
Need to add Parameters to describe `input` in docstring.  Also, `input` is a python command and should not be used as a variable name.